src/main/java/bio/terra/controller/DataRepositoryServiceApiController.java

```diff
@@ -0,0 +1,134 @@
+package bio.terra.controller;
+
+import bio.terra.configuration.ApplicationConfiguration;
+import bio.terra.controller.exception.ValidationException;
+import bio.terra.model.DRSAccessURL;
+import bio.terra.model.DRSObject;
+import bio.terra.model.DRSServiceInfo;
+import bio.terra.service.DatasetService;
+import bio.terra.service.DrsService;
+import bio.terra.service.FileService;
+import bio.terra.service.JobService;
+import bio.terra.service.SamClientService;
+import bio.terra.service.StudyService;
+import bio.terra.validation.DatasetRequestValidator;
+import bio.terra.validation.IngestRequestValidator;
+import bio.terra.validation.StudyRequestValidator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Optional;
+
+@Controller
+public class DataRepositoryServiceApiController implements DataRepositoryServiceApi {
+
```

Comment:
No, this is derived from a swagger-generated controller. It looks similar because I copied RepositoryApiController. As I've worked on it more, it has gotten a good bit different, but that is in the next PR...